src/publisher.ts

```diff
@@ -256,11 +266,12 @@ export class Publisher {
         },
         (err, resp) => {
           const messageIds = arrify(resp && resp.messageIds);
-          each(callbacks, (callback, next) => {
-            const messageId = messageIds[callbacks.indexOf(callback)];
-            callback(err, messageId);
-            next();
-          });
+          each(
+              callbacks, (callback: PublishCallback, next: PublishCallback) => {
```

Comment:
Oof, missed this, but `next` should probably just be `Function`.

Reply:
done